File: datasets/jl-keep-code_davinci_001_temp_0.2-0/HumanEval_12_longest.jl
""" Out of list of strings, return the longest one. Return the first one in case of multiple
    strings of the same length. Return None in case the input list is empty.
    >>> longest([])

    >>> longest(['a', 'b', 'c'])
    'a'
    >>> longest(['a', 'bb', 'ccc'])
    'ccc'
    """
function longest(strings::Vector{String})::Union{String, Nothing} 
    if isempty(strings)
        return nothing
    end
    longest_string = strings[1]
    for string in strings
        if length(string) > length(longest_string)
            longest_string = string
        end
    end
    return longest_string
end

using Test

@testset begin

candidate = longest;
	@test(candidate(Vector{String}([])) == nothing)
	@test(candidate(["x", "y", "z"]) == "x")
	@test(candidate(["x", "yyy", "zzzz", "www", "kkkk", "abc"]) == "zzzz")
end

File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_12_longest.rb
# Out of list of strings, return the longest one. Return the first one in case of multiple
# strings of the same length. Return None in case the input list is empty.
# >>> longest([])
# >>> longest(['a', 'b', 'c'])
# 'a'
# >>> longest(['a', 'bb', 'ccc'])
# 'ccc'
def longest(strings)
  strings.max_by(&:length)
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_longest
    candidate = method(:longest)
    assert_equal(nil, candidate.call([]))
    assert_equal("x", candidate.call(["x", "y", "z"]))
    assert_equal("zzzz", candidate.call(["x", "yyy", "zzzz", "www", "kkkk", "abc"]))
  end
end
